src/common/components/select-repository-list/index.js

```diff
@@ -0,0 +1,145 @@
+                            `?name=${encodeURIComponent(snapName)}`;
+
+    return <span>
+      The name provided in the snapcraft.yaml file ({snapName}) is not
+      registered in the store.
+      Please <a href={registerNameUrl}>register it</a> before trying
+      again.
+    </span>;
+  }
+
+  getErrorMessage(error) {
+    if (error) {
+      const payload = error.json.payload;
+      if (payload.code === SNAP_NAME_NOT_REGISTERED_ERROR_CODE) {
+        return this.getSnapNotRegisteredMessage(payload.snap_name);
+      } else {
+        return error.message;
+      }
+    }
+  }
+
+  renderRepository(repo) {
+    const { fullName } = repo;
+    const status = this.props.repositoriesStatus[fullName] || {};
+    const { selectedRepos } = this.props.selectRepositoriesForm;
+
+    return (
+      <SelectRepositoryRow
+        key={ `repo_${repo.fullName}` }
+        repository={ repo }
+        buttonLabel={ status.isFetching ? 'Creating...' : 'Create' }
+        buttonDisabled={ status.isFetching }
+        onChange={ this.onSelectRepository.bind(this, repo) }
+        errorMsg= { this.getErrorMessage(status.error) }
+        checked={ hasRepository(selectedRepos, repo) }
+      />
+    );
+  }
+
+  onSelectRepository(repository) {
+    this.props.dispatch(toggleRepository(repository));
+  }
+
+  onSubmit() {
+    const { selectedRepos } = this.props.selectRepositoriesForm;
+    if (selectedRepos.length) {
+      // TODO: Switch to batched repository action
+      this.props.dispatch(createSnap(selectedRepos[0].url));
+    }
+  }
+
+  onPageLinkClick(pageNumber) {
+    this.props.dispatch(fetchUserRepositories(pageNumber));
+  }
+
+  render() {
+    const isLoading = this.props.repositories.isFetching;
+    const pageLinks = this.props.repositories.pageLinks;
+
+    return (
+      <div>
+        { this.props.repositories.success && pageLinks &&
+          <div className={ styles['page-links-container'] }>
+            <PageLinks { ...pageLinks } onClick={ this.onPageLinkClick.bind(this) } />
+          </div>
+        }
+        { isLoading &&
+          <div className={ spinnerStyles }><Spinner /></div>
+        }
+        { this.props.repositories.success &&
+          this.props.repositories.repos.map(this.renderRepository.bind(this))
+        }
+        { this.props.repositories.success && pageLinks &&
+          <div className={ styles['page-links-container'] }>
+            <PageLinks { ...pageLinks } onClick={ this.onPageLinkClick.bind(this) } />
+          </div>
+        }
+        <div className={ styles.footer }>
+          <div className={ styles.left }>
+            <HeadingThree>
+              In order to enable your repositories, you need to sign in to your Ubuntu One account.
+            </HeadingThree>
+          </div>
+          <div className={ styles.right }>
+            <Button onClick={ this.onSubmit.bind(this) } appearance={ 'positive' }>
+              Enable repos
```

Comment:
I don't know if this is in the copy from design, but personally I'd prefer that we not abbreviate "repositories" to "repos" outside of casual sketches.